troughs_and_waves/gaussian/tests/test_gaussian.py

```diff
@@ -0,0 +1,51 @@
+#  (C) Copyright 2010-2020 Enthought, Inc., Austin, TX
+#  All rights reserved.
+
+import unittest
+
+from unittest import mock
+
+from traits.testing.api import UnittestTools
+from force_bdss.api import DataValue, Slot, BaseDataSourceFactory
+
+from troughs_and_waves.gaussian.gaussian import Gaussian
+from troughs_and_waves.gaussian.gaussian_model import GaussianModel
+
+
+class TestGaussian(unittest.TestCase, UnittestTools):
+    def setUp(self):
+        self.factory = mock.Mock(spec=BaseDataSourceFactory)
```

Comment:
If we import `GaussianDataSourceFactory` here, we can change this assignment to:

```python
self.factory = GaussianDataSourceFactory(
    plugin={'id':'0', 'name': 'test'})
```

In order to test the factory functionalities alongside our other data source tests. This works because the `plugin` argument for `BaseFactory.__init__` can either be an `envisage.Plugin` instance or just a dictionary with `'id'` and `'name'` keys.